homeassistant/components/velbus/entity.py

```diff
@@ -26,16 +32,33 @@ def __init__(self, channel: VelbusChannel) -> None:
         self._attr_name = channel.get_name()
         self._attr_device_info = DeviceInfo(
             identifiers={
-                (DOMAIN, str(channel.get_module_address())),
+                (DOMAIN, self._get_identifier()),
             },
             manufacturer="Velleman",
             model=channel.get_module_type_name(),
+            model_id=str(channel.get_module_type()),
             name=channel.get_full_name(),
             sw_version=channel.get_module_sw_version(),
+            serial_number=channel.get_module_serial(),
+            via_device=self._get_via_device(),
         )
         serial = channel.get_module_serial() or str(channel.get_module_address())
         self._attr_unique_id = f"{serial}-{channel.get_channel_number()}"
 
+    def _get_identifier(self) -> str:
+        """Return the identifier of the entity."""
+        if not self._channel.is_sub_device():
+            return str(self._channel.get_module_address())
+        return (
+            f"{self._channel.get_module_address()}-{self._channel.get_channel_number()}"
+        )
+
+    def _get_via_device(self) -> tuple[str, str]:
+        """Return the via device tuple."""
+        if self._channel.is_sub_device():
+            return (DOMAIN, str(self._channel.get_module_address()))
+        return ("", "")
```

Comment:
I'd rather not return this, doesn't via_device accept `None`?

Reply:
looking here it only accepts a tuple ...

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/device_registry.py#L105

or am i missing something?